packages/graphql_server/src/starHistory/forkHistory.ts

```diff
@@ -0,0 +1,302 @@
+ * @param {string} repo - Name of the Github repository: "owner/repository"
+ * @param {string} token - Github access token
+ * @returns Promise<number> : A promise that resolves to the total count of forks for the repository
+ */
+async function getRepoForksCount(repo: string, token?: string): Promise<number> {
+  const response: AxiosResponse<{ forks_count: number }> = await axios.get(
+    `https://api.github.com/repos/${repo}`,
+    {
+      headers: {
+        Accept: 'application/vnd.github+json',
+        Authorization: token ? `token ${token}` : ''
+      }
+    }
+  )
+
+  return response?.data?.forks_count
+}
+
+/** Retrieves the fork records (fork count by date) of a Github repository
+ * @param {string} repo - Name of the Github repository: "owner/repository"
+ * @param {string} token - Github Access token
+ * @param {number} maxRequestAmount - Maximum number of API requests to make to retrieve the forks
+ * The higher this value is the more accurate is going to be the graph of the fork history
+ * @param {number} startPage - a possible start page for the partial history
+ * @returns Promise<Array<{ date: string, count: number }>>: A promise that resolves to an array of fork records
+ */
+export async function getRepoForkRecords(
+  repo: string,
+  token: string,
+  maxRequestAmount: number,
+  startPage?: number
+) {
+  const pageCount = await getPageCount(repo, token)
+
+  const requestPages: number[] = []
+  if (startPage == undefined) {
+    if (pageCount < maxRequestAmount) {
+      requestPages.push(...utils.range(1, pageCount))
+    } else {
+      utils.range(1, maxRequestAmount).forEach((i: number) => {
+        requestPages.push(Math.round((i * pageCount) / maxRequestAmount) - 1)
+      })
+      if (!requestPages.includes(1)) {
+        requestPages.unshift(1)
+      }
+    }
+  } else {
+    if (pageCount < maxRequestAmount) {
+      requestPages.push(...utils.range(pageCount - startPage, pageCount))
+    } else {
+      utils.range(1, maxRequestAmount).forEach((i: number) => {
+        requestPages.push(
+          Math.round(pageCount - startPage + (i * startPage) / maxRequestAmount) - 1
+        )
+      })
+    }
+  }
+
+  return await getRepoForksMap(repo, token, requestPages, maxRequestAmount)
+}
+
+/** Retrieves the page count for the forks of a GitHub repository.
+ * @param {string} repo - Name of the GitHub repository in the format "owner/repository".
+ * @param {string} token - GitHub access token.
+ * @returns {Promise<number>} A promise that resolves to the total number of pages.
+ * @throws {object} Throws an error object if the request fails or the repository has no forks.
+ */
+async function getPageCount(repo: string, token: string) {
+  const patchRes: AxiosResponse<ForksData[]> = await getRepoForks(repo, token)
+
+  const headerLink: string = (patchRes.headers['link'] as string) || ''
+
+  let pageCount = 1
+  const regResult = /next.*&page=(\d*).*last/.exec(headerLink)
```

Comment:
An explanatory comment for RegEx explanations would be nice. Something like:
`// Regex to extract the 'next' and 'last' page numbers from GitHub's pagination Link header`

Also, it looks like you are only using regResult[1], so you might as well make it:
`const lastPageNumber = /next.*&page=(\d*).*last/.exec(headerLink)[1]` :)

Reply:
@ArnoClaude Thanks for the already written out explanation. I will implement it in the next commit